src/xen/xen.c

```diff
@@ -379,6 +387,33 @@ closure_function(1, 0, void, xen_per_cpu_init,
     xen_setup_vcpu(current_cpu()->id, bound(shared_info_phys));
 }
 
+void unix_shutdown(void);
+
+define_closure_function(1, 0, void, xen_shutdown_handler,
+                 buffer, b)
+{
+    buffer rsp = bound(b);
+    buffer_clear(rsp);
+    u32 txid;
+    status s = xenstore_transaction_start(&txid);
+    if (!is_ok(s))
+        return;
+    s = xenstore_read_string(txid, alloca_wrap_cstring("control"), "shutdown", rsp);
```

Comment:
Interestingly, gcc 11 doesn't complain about this one, but does complain about uninitialized result in xenstore_read_u64 below, which is why result is initialized there. I added tx_id initialization in xenstore_transaction_start instead of changing its callers.